src/shell/shell_cmds.cpp

```diff
@@ -465,26 +472,250 @@ static std::string to_search_pattern(const char *arg)
 	char buffer[CROSS_LEN];
 	pattern = ExpandDot(pattern.c_str(), buffer, sizeof(buffer));
 
-	// When there's no wildcard and target is a directory then search files
-	// inside the directory.
-	const char *p = pattern.c_str();
-	if (!strrchr(p, '*') && !strrchr(p, '?')) {
-		uint16_t attr = 0;
-		if (DOS_GetFileAttr(p, &attr) && (attr & DOS_ATTR_DIRECTORY))
-			pattern += "\\*.*";
+	return pattern;
+}
+
+uint32_t byte_count,file_count,dir_count;
+Bitu p_count;
+std::vector<std::string> dirs, adirs;
+
+static size_t GetPauseCount() {
+	return (real_readb(BIOSMEM_SEG,BIOSMEM_NB_ROWS) > 2u) ? (real_readb(BIOSMEM_SEG,BIOSMEM_NB_ROWS)) : 24u;
+}
```

Comment:
Can a comment be added describing what PauseCount means, and what the bounds checks at 2u and 24u mean?

``` c++
static size_t GetPauseCount() {
	const uint8_t pause_count = real_readb(BIOSMEM_SEG,BIOSMEM_NB_ROWS) ;
	return pause_count > 2u ? pause_count : 24u;
}

Reply:
@kcgen The current code before this PR simply fixes it to 24, which I don't think it is a good idea. As a result, the new code will read the actual number of rows in the screen instead, but if this value is deemed to be totally incorrect (0, 1, 2) then it will still return 24 as the value, otherwise it will be accepted as is. Of course this limit (i.e. 2) can be changed, but it is just a matter of convenience and can be for example replaced by assert of course.

Reply:
If 3 if the lowest acceptable sane value, then I would suggest returning 3 for the 0, 1 and 2 cases.

Reply:
@kcgen I have modified the code so that if the read value is 0 or >100, then they are simply treated as invalid values and return 24 as before. On the other hand, if the read value is 1 or 2, then return 3 as you suggested.